### ISSUE TITLE
dotnet format tries to load unreleased package Microsoft.CodeAnalysis 4.6.0.0

### ISSUE THREAD

**Issue by korDen** [dotnet/format#1834]
Freshly installed Visual Studio 2022 Preview.

dotnet add package Microsoft.CodeAnalysis installs version 4.5
dotnet add package Microsoft.CodeAnalysis -v 4.6 FAILS with

error: NU1102: Unable to find package Microsoft.CodeAnalysis with version (>= 4.6.0)
error: - Found 141 version(s) in nuget.org [ Nearest version: 4.6.0-1.final ]
error: - Found 0 version(s) in Microsoft Visual Studio Offline Packages
error: Package 'Microsoft.CodeAnalysis' is incompatible with 'all' frameworks in project 'C:\Users\korDen\source\repos\WebApplication1\WebApplication1\WebApplication1.csproj'.

Somehow 4.6.0-1.final is NOT considered to be >= 4.6.0
(same with dotnet add package Microsoft.Net.Compilers.Toolset)

Installing 4.6.0-1.final doesn't help either, dotnet format still fails with

Unhandled exception: System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.
Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

**Comment by korDen** [comment]
$ dotnet format --version
7.0.357101+49c2ef651359526841d13e66129b71d1bcd9cef9

$ dotnet --info
.NET SDK:
 Version:   7.0.300-preview.23179.2
 Commit:    6d2b42e30e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.300-preview.23179.2\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  7.0.203 [C:\Program Files\dotnet\sdk]
  7.0.300-preview.23179.2 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by trampster** [comment]
We can reproduce this in our CI builds in docker and on your ubuntu dev machines. We rolled back our docker image to before it broke, but we can't figure out how to go to an older version on our linux desktops using the microsoft PPA

**Comment by adam-knights** [comment]
I had a quick flick through recent code changes in areas touching the version bits around CodeAnalysis and there have been recent improvements and fix ups for merge conflicts etc, @MichaelSimons and @ViktorHofer might be able to quickly see what is causing this? This seems to be a fairly big regression affecting lots of companies as they take .NET 7.0.302 - see https://github.com/dotnet/sdk/issues/32598 .

**Comment by ViktorHofer** [comment]
Just verified that this is unrelated to any of the recently made changes as dotnet/format's main branch doesn't flow into sdk's release/7.x branches.

**Comment by robindv** [comment]
> We can reproduce this in our CI builds in docker and on our ubuntu dev machines. We rolled back our docker image to before it broke, but we can't figure out how to go to an older version on our linux desktops using the microsoft PPA

I was able to downgrade by changing the priority of this version creating a file `/etc/apt/preferences.d/dotnet-sdk`:
```
Package: dotnet-sdk-7.0
Pin: version 7.0.302-1
Pin-Priority: 99
```

After that I removed and reinstalled the package and that installed 7.0.203

**Comment by glen-84** [comment]
Effectively a duplicate of #1800.

See also https://github.com/dotnet/sdk/issues/32598 and https://github.com/dotnet/format/pull/1851.

**Comment by glen-84** [comment]
@korDen Is this fixed for you in 7.0.304? If so, please close this issue.

**Comment by sharwell** [comment]
Fixed by #1885
